### PR TITLE
Some clang fixes

### PR DESCRIPTION
**Pull request by jcs**

### CODE REVIEW

File: AK/OwnPtr.h [hunk 1]
```diff
@@ -204,7 +204,12 @@ inline void swap(OwnPtr<T>& a, OwnPtr<U>& b)
 template<typename T>
 struct Traits<OwnPtr<T>> : public GenericTraits<OwnPtr<T>> {
     using PeekType = const T*;
-    static unsigned hash(const OwnPtr<T>& p) { return int_hash((u32)p.ptr()); }
+    static unsigned hash(const OwnPtr<T>& p) {
+        if constexpr(sizeof(uintptr_t) == 8)
+            return u64_hash((u64)p.ptr());
+        else
+            return int_hash((u32)p.ptr());
```
Comment (awesomekling) [inline AK/OwnPtr.h]: Maybe we could add a `ptr_hash` function that takes care of this internally?

File: Libraries/LibGfx/Forward.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ class Point;
 class Rect;
 class Size;
 class StylePainter;
-class SystemTheme;
+struct SystemTheme;
```
Comment (bugaevc) [inline Libraries/LibGfx/Forward.h]: > SystemTheme.h:81:1: error: 'SystemTheme' defined as a struct here but previously declared as a class; this is valid, but may result in linker errors under the Microsoft C++ ABI

Yeah I guess we shouldn't risk getting linker errors under the Microsoft C++ ABI :smiley:

Comment (jcs) [reply Libraries/LibGfx/Forward.h]: Yeah I wouldn't care either but it blocks compilation with `-Werror` on.

Comment (bugaevc) [reply Libraries/LibGfx/Forward.h]: No, I get it, just found it somewhat funny 😀

**Comment by jcs** [comment]
This adds `ptr_hash` but it fails compiling `NotificationServer`.  I'm not sure if the fault is in `NotificationServer` or if the `hash()` functions are expected to cast to a `uintptr_t` before calling `ptr_hash`.

**Comment by awesomekling** [comment]
Oops, sorry I forgot about this one! Thank you for the fixes :D

**Pull request merged**